synapse/handlers/auth.py

```diff
@@ -558,27 +549,66 @@ def _auth_dict_for_flows(
                     params[stage] = get_params[stage]()
 
         return {
-            "session": session["id"],
+            "session": session_id,
             "flows": [{"stages": f} for f in public_flows],
             "params": params,
         }
 
-    def _get_session_info(self, session_id: Optional[str]) -> dict:
+    def _create_session(
+        self,
+        clientdict: Dict[str, Any],
+        ui_auth: Tuple[str, str, Dict[str, Any]],
+        description: str,
+    ) -> dict:
```

Comment:
Might want to mention that this method modifies `self.sessions` itself.

Reply:
I'm not sure I follow why that's important to mention in the comment, theoretically you shouldn't care where the sessions are stored. (`sessions` should really be private, by the way.)

Reply:
It's just that it may be easy for someone to use this method, thinking it just creates a session, and then they themselves puts it into `self.sessions` to make sure it gets saved.

Though this is probably just a review-isolation think. Actually looking throughout the rest of the class, I think one gets the impression that `self.session` is managed by the helper functions.

So this probably isn't a problem.

Reply:
Yeah, I see what you're saying. FWIW the `sessions` dict goes away in #7302 and is replaced by a proper store.